src/Microsoft.ML.Transforms/TensorFlow/TensorflowUtils.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.ML.Runtime.Data;
+
+namespace Microsoft.ML.Transforms.TensorFlow
+{
+    internal partial class TensorflowUtils
+    {
+        internal static PrimitiveType Tf2MlNetType(TFDataType type)
+        {
+            switch (type)
+            {
+                case TFDataType.Float:
+                    return NumberType.R4;
+                case TFDataType.Double:
+                    return NumberType.R8;
+                case TFDataType.Int32:
+                    return NumberType.I4;
+                case TFDataType.Int64:
+                    return NumberType.I8;
+                case TFDataType.UInt32:
+                    return NumberType.U4;
+                case TFDataType.UInt64:
+                    return NumberType.U8;
+                case TFDataType.Bool:
+                    return BoolType.Instance;
+                case TFDataType.String:
+                    return TextType.Instance;
+                default:
+                    throw new NotSupportedException("Tensorflow type not supported.");
+            }
+        }
+
+        public static unsafe void FetchData<T>(IntPtr data, T[] result)
+        {
+            var size = result.Length;
+
+            GCHandle handle = GCHandle.Alloc(result, GCHandleType.Pinned);
+            IntPtr target = handle.AddrOfPinnedObject();
+
+            Int64 sizeInBytes = size * Marshal.SizeOf((typeof(T)));
+            Buffer.MemoryCopy(data.ToPointer(), target.ToPointer(), sizeInBytes, sizeInBytes);
+            handle.Free();
+        }
+
+        internal static bool IsTypeSupported(TFDataType tfoutput)
```

Comment:
>IsTypeSupported [](start = 29, length = 15)

These are for input types, we should decide whether we'd like to support other types as well.

Reply:
@yaeld@microsoft.com .. Should we track this as a separate task in the GitHub board we are using ?

---
In reply to: [212126644](https://github.com/dotnet/machinelearning/pull/704#discussion_r212126644) [](ancestors = 212126644)